File: backend/fusion/temporal_gate.py
from collections import deque
import numpy as np

class TemporalGate:
    def __init__(self, window=5, threshold=0.6):
        self.buffer = deque(maxlen=window)
        self.threshold = threshold

    def update(self, score):
        self.buffer.append(score)

    def is_consistent(self):
        if len(self.buffer) < self.buffer.maxlen:
            return False
        return np.mean(self.buffer) >= self.threshold

File:  android-app/app/src/main/java/com/example/safeguard/ml/TemporalGate.java
package com.example.safeguard.ml;

import java.util.ArrayDeque;

public class TemporalGate {

    private ArrayDeque<Float> buffer = new ArrayDeque<>();
    private int window = 5;
    private float threshold = 0.6f;

    public boolean update(float score) {
        if (buffer.size() == window) buffer.poll();
        buffer.add(score);

        if (buffer.size() < window) return false;

        float sum = 0f;
        for (float s : buffer) sum += s;
        return (sum / window) >= threshold;
    }
}
